hadoop/Dockerfile

```diff
@@ -1,36 +1,41 @@
 # syntax=docker/dockerfile:1
-FROM stackable/image/java-base
+FROM stackable/image/java-base AS builder
 
 ARG PRODUCT
-ARG RELEASE
-
-LABEL name="Apache Hadoop" \
-      maintainer="info@stackable.tech" \
-      vendor="Stackable GmbH" \
-      version="${PRODUCT}" \
-      release="${RELEASE}" \
-      summary="The Stackable image for Apache Hadoop." \
-      description="This image is deployed by the Stackable Operator for Apache Hadoop / HDFS."
+ARG JMX_EXPORTER_VERSION=0.18.0
 
 # https://github.com/hadolint/hadolint/wiki/DL4006
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
+# unzip & zip are required for log4shell.sh
+# All others are required for the FUSE build
 RUN microdnf update && \
-    microdnf install tar gzip zip krb5-workstation openssl && \
+    microdnf install \
+    cmake \
+    cyrus-sasl-devel \
+    fuse-devel \
+    gcc \
+    gcc-c++ \
+    java-11-openjdk-devel \
+    maven \
+    openssl-devel \
+    tar \
+    unzip \
+    zip && \
     microdnf clean all
 
-COPY --chown=stackable:stackable hadoop/stackable /stackable
-COPY hadoop/licenses /licenses
-
-USER stackable
 WORKDIR /stackable
 
-RUN curl -L https://repo.stackable.tech/repository/packages/hadoop/hadoop-${PRODUCT}.tar.gz | tar -xzC . && \
+# This is needed here because it creates the JMX directory, we could create it any other way but this works
+COPY hadoop/stackable /stackable
+
+RUN curl --fail -L https://repo.stackable.tech/repository/packages/hadoop/hadoop-${PRODUCT}-src.tar.gz | tar -xzC . && \
+    curl --fail -L https://repo.stackable.tech/repository/packages/hadoop/hadoop-${PRODUCT}.tar.gz | tar -xzC . && \
     ln -s /stackable/hadoop-${PRODUCT} /stackable/hadoop && \
-    ln -sf /stackable/bin/start-namenode /stackable/hadoop/bin && \
-    curl https://repo.stackable.tech/repository/packages/jmx-exporter/jmx_prometheus_javaagent-0.16.1.jar \
-    -o /stackable/jmx/jmx_prometheus_javaagent-0.16.1.jar && \
-    chmod -x /stackable/jmx/jmx_prometheus_javaagent-0.16.1.jar
+    rm -rf /stackable/hadoop/lib/native/examples && \
+    rm -rf /stackable/hadoop/share/doc && \
+    curl --fail https://repo.stackable.tech/repository/packages/jmx-exporter/jmx_prometheus_javaagent-${JMX_EXPORTER_VERSION}.jar -o /stackable/jmx/jmx_prometheus_javaagent-${JMX_EXPORTER_VERSION}.jar && \
```

Comment:
Can we do a link like this? That means we can stop hardcoding the version in the operators.
https://github.com/stackabletech/docker-images/blob/109af74441b0ba6ad7e784d020903e0e1bbd1c12/trino/Dockerfile#L35C1-L38C114

Reply:
Good idea. I added it.

Trino calls it `JMX_EXPORTER` while I call it `JMX_EXPORTER_VERSION`
Any preferences?

Reply:
Do I then also need to change the operator?

Reply:
Answer: Yes. I found it and will add it.

Reply:
> Good idea. I added it.
> 
> Trino calls it `JMX_EXPORTER` while I call it `JMX_EXPORTER_VERSION` Any preferences?

I like JMX_EXPORTER_VERSION better.

> Do I then also need to change the operator?

This would be somewhat breaking. It will only work with new images, not old ones. 
I just wanted to have it in every operator and migrate at some point.

Reply:
I opted to use `JMX_EXPORTER`.
It looks better in conf.py and it is already used in Trino. I wanted to stay consistent.